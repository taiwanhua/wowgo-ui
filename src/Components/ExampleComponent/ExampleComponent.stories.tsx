import * as React from "react";
import { Story, Meta } from '@storybook/react/types-6-0';
import { ExampleComponent, ExampleComponentProps } from "./ExampleComponent";

export default {
    title: 'Example/ExampleComponent',
    component: ExampleComponent,
    argTypes: {
        backgroundColor: { control: 'color' },
    },
} as Meta;


const Template: Story<ExampleComponentProps> = (args) => <ExampleComponent {...args} />;

export const Primary = Template.bind({});
Primary.args = {
    title: "hello world ~",
};

export const Secondary = Template.bind({});
Secondary.args = {
    title: "hello !! "
};

export const Large = Template.bind({});
Large.args = {
    title: "Welcome !! "
};
