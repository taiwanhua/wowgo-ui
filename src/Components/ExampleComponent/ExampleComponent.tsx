import * as React from "react";
import styled from "styled-components";


// import original module declarations
// import 'styled-components';

// // and extend them!
// declare module 'styled-components' {
//     export interface DefaultTheme {
//         borderRadius: string;

//         colors: {
//             main: string;
//             secondary: string;
//         };
//     }
// }

export interface BaseProps {
    className?: string;
    disabled?: boolean;
}

export interface TitleProps extends BaseProps {
    backgroundColor?: string;
    title: string;
}

export interface TitleWithH1Props extends BaseProps {
    backgroundColor?: string;
}

const TitleWithH1 = styled.h1.attrs<TitleWithH1Props>((props) => ({ ...(props.disabled && { disabled: true }) }))`
    background-color: ${(props: TitleWithH1Props) => (props.backgroundColor ?? "#d4d4d4")};
`;


// const TitleWith = styled((props: TitleProps) => (
//     <h1 className={props.className}>
//         {props.title}
//     </h1>
// )).attrs<TitleProps>((props) => ({ ...(props.disabled && { disabled: true }) }))`
//     background-color: ${(props: TitleProps) => (props.backgroundColor ?? "#d4d4d4")};
// `;



export interface ExampleComponentProps {
    /**
     * ExampleComponentProps 的文字
     */
    title: string

    /**
     * disabled 參數
     */
    disabled?: string

    /**
     * ExampleComponentProps 的背景顏色
     */
    backgroundColor: string
}

export const ExampleComponent: React.FC<ExampleComponentProps> = (props) => {

    return (
        <React.Fragment>
            {/* <TitleWith title={props?.title} backgroundColor={props.backgroundColor} /> */}
            <TitleWithH1 backgroundColor={props.backgroundColor} >{props?.title}</TitleWithH1>
        </React.Fragment>
    )
}