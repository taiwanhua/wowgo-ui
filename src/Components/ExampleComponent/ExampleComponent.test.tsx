import * as React from "react";
import { shallow } from "enzyme";
import { ExampleComponent } from "./ExampleComponent";
it("Renders without crashing", () => {
    const wrapper = shallow(<ExampleComponent title={"Test"} backgroundColor={"red"} />);
    expect(wrapper).toMatchSnapshot();
});