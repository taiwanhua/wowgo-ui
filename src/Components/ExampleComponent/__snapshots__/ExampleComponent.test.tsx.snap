// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<Fragment>
  <styled.h1
    backgroundColor="red"
  >
    Test
  </styled.h1>
</Fragment>
`;
